python-frontend/src/test/java/org/sonar/python/tree/PythonTreeMakerTest.java

```diff
@@ -1779,6 +1780,14 @@ public void string_interpolation() {
     assertThat(((Name) nestedInterpolation).name()).isEqualTo("x");
     assertThat(nestedInterpolation.firstToken().line()).isEqualTo(2);
     assertThat(nestedInterpolation.firstToken().column()).isEqualTo(15);
+
+    // interpolated expression contains curly braces
+    stringLiteral = (StringLiteral) parse("f'{ {element for element in [1, 2]} }'", treeMaker::expression);
+    assertThat(stringLiteral.stringElements()).hasSize(1);
+    elmt = stringLiteral.stringElements().get(0);
+    assertThat(elmt.isInterpolated()).isTrue();
+    assertThat(elmt.interpolatedExpressions()).hasSize(1);
+    assertThat(elmt.interpolatedExpressions().get(0).is(Kind.COMP_FOR));
```

Comment:
I think you forgot `isTrue()` here. And in this case the kind should be "SET_COMPREHENSION"